lab-steven

BatemanVO opened the pull request.

BatemanVO committed:
Initial scaffold.

BatemanVO committed:
Created initial server file.

BatemanVO committed:
Updated package.json with npm scripts.

BatemanVO committed:
Replaced debug with console.log in app.listen.

BatemanVO committed:
Created student constructor.

BatemanVO committed:
Updated constructor with debug and error handling.

BatemanVO committed:
Added createItem method to storage.js.

BatemanVO committed:
Added getItem method to storage.js.

BatemanVO committed:
Added updateItem method to storage.js.

BatemanVO committed:
Added getAllItems method to storage.js.

BatemanVO committed:
Added getStudent method to Student constructor.

BatemanVO committed:
Added readStudent and updateStudent method to Student constructor.

BatemanVO committed:
Added readAllStudents method to Student constructor.

BatemanVO committed:
Created error middleware.

BatemanVO committed:
Created cors middleware.

BatemanVO committed:
Required in new middleware in server.js.

BatemanVO committed:
Added router file to route directory.

BatemanVO committed:
Exported and required studentRouter.

BatemanVO committed:
Fixes to updateItem.
In student-router.js, passed request.body in as  additional argument to updateStudent.
In student-constructor.js, passed in string of schemaName.
In storage.js, added return statements to the readItem and createItem methods.

BatemanVO committed:
Added tests for createStudent.

BatemanVO committed:
Fixed unterminated quote in npm test.

BatemanVO committed:
Required server in for tests.

BatemanVO committed:
Removed unnecessary after hook in invalid post test.

BatemanVO committed:
Created tests for GET routes.

BatemanVO committed:
Fixed no ID test.
Added before/after to create and delete a student. Also changed expects to look for an array and to look at length.

BatemanVO committed:
Added tests for PUT routes.

BatemanVO committed:
Minor corrections.
In the test, changed sampleStudent.id to this.tempStudent.id to fix undefined.
In storage, changed (not-content) to Object.keys(content).length < 1 to account for empty objects not technically causing errors.

BatemanVO committed:
Added tests for DELETE routes.

BatemanVO committed:
Fixed minor issues.
Changed error thrown from delete with wrong ID to 404 error.
Added a missing done() to test.

BatemanVO committed:
Added README file.

giodamelio commented on lab-steven/server.js:
```diff
@@ -0,0 +1,18 @@
+
+app.use(morgan('dev'));
+app.use(studentRouter);
+app.use(cors);
```
Your cors middleware should be before your api route, otherwise the headers won't be applied before the response is sent.

BatemanVO committed:
Refactored tests to make use of beforeEach and afterEach.

BatemanVO committed:
Moved app.use(cors) call.
Cors needs to be included before the router or else the headers won't be applied before the response is set.
